### DISCUSSION TITLE
Rapidly increasing number of IKE SAs

### DISCUSSION THREAD

**Discussion by enjoycowboy** [strongswan/strongswan#1067] [Q&A]
I have a vpn gateway-type device running strongswan handling around 1000 host to host tunnels and over time the number of SAs negotiated between peers rise to levels that cannot be explained, given that the tunnels connect a single machine to another single machine on the other end. Also, the number of jobs scheduled by StrongSwan to match the endless spawning of SAs increases nonstop, and it all comes to a head every other day with StrongSwan just crashing out and being revived by systemd. we then have to manually go and restart the VPN on the other ends. Here is the output of swanctl --stats:

```
uptime: 21 hours, since May 25 20:05:06 2022
worker threads: 16 total, 11 idle, working: 4/0/1/0
job queues: 0/0/0/0
jobs scheduled: 73329
IKE_SAs: 7803 total, 15 half-open
mallinfo: sbrk 290586624, mmap 0, used 152415392, free 138171232
loaded plugins: charon-systemd pkcs11 tpm aesni aes des blowfish
 rc2 sha2 sha3 sha1 md4 md5 mgf1 rdrand random nonce x509
 revocation constraints acert pubkey pkcs1 pkcs7 pkcs8 pkcs12 pgp
 dnskey sshkey pem openssl gcrypt af-alg fips-prf gmp curve25519 
chapoly xcbc cmac hmac ctr ccm gcm drbg newhope curl sqlite attr 
kernel-netlink resolve socket-default connmark farp stroke vici updown
 eap-identity eap-sim eap-aka eap-aka-3gpp eap-aka-3gpp2 eap-md5 
eap-gtc eap-mschapv2 eap-dynamic eap-radius eap-tls eap-ttls 
eap-peap eap-tnc xauth-generic xauth-eap xauth-pam xauth-noauth 
tnc-imc tnc-imv tnc-tnccs tnccs-20 tnccs-11 tnccs-dynamic dhcp led unity counters

```

Here is some more info:
[host to host, remote peer behind ddns](https://github.com/strongswan/strongswan/files/8789201/tunnel74.txt)
[host to host, remote peer behind dynamic IP](https://github.com/strongswan/strongswan/files/8789202/tunnel86.txt)
[host to host, fixed IP](https://github.com/strongswan/strongswan/files/8789203/tunnel125.txt)
[tunnel_common](https://github.com/strongswan/strongswan/files/8789204/tunnel_common.txt)
[charon.log](https://github.com/strongswan/strongswan/files/8789224/vpn-ipsec-log.zip)
[the output of swanctl -L](https://github.com/strongswan/strongswan/files/8789227/list-sas.zip)

**Comment by tobiasbrunner** [comment]
Note that `closeaction=restart` might not be a good choice depending on whether the other end uses break-before-make reauthentication (see [the docs](https://docs.strongswan.org/docs/5.9/config/rekeying.html)).
